### PR TITLE
Adding  windows cuda 11.5 workflows

### LINKED ISSUE #69081: Add CUDA 11.5 to install to windows AMI

### PR DESCRIPTION
**Pull request by atalman**
Fixes #69081

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -909,6 +910,11 @@ def get_selected_tests(options):
             WINDOWS_BLOCKLIST.append("jit")
             WINDOWS_BLOCKLIST.append("jit_fuser")
 
+        if distutils.version.LooseVersion(torch.version.cuda) >= "11.5":
```
Comment (janeyx99) [inline test/run_test.py:913]: Is there an issue tracking why this is the case? Could you add a comment above here explaining why this has to be/linking to the issue?

Comment (atalman) [reply test/run_test.py:913]: Yes totally The is the issue https://github.com/pytorch/pytorch/issues/69460 Will add a comment as well.

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -293,6 +293,19 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
...
+        arch="windows",
+        build_environment="win-vs2019-cuda11.5-py3",
+        cuda_version="11.5",
+        test_runner_type=WINDOWS_CUDA_TEST_RUNNER,
+        num_test_shards=2,
+        only_run_smoke_tests_on_pull_request=True,
+        enable_force_on_cpu_test=1,
+        ciflow_config=CIFlowConfig(
+            run_on_canary=True,
+            labels={LABEL_CIFLOW_DEFAULT, LABEL_CIFLOW_CUDA, LABEL_CIFLOW_WIN}
```
Comment (seemethere) [inline .github/scripts/generate_ci_workflows.py:306]: Can we make this a periodic job as well? And remove LABEL_CIFLOW_DEFAULT from here?

Comment (atalman) [reply .github/scripts/generate_ci_workflows.py:306]: Ah yes totally let me do this

**Review by malfet — APPROVED** [review]

**Review by janeyx99 — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

**Comment by seemethere** [comment]
Unfortunately looks like this was causing upstream windows failures:

```
(base) C:\actions-runner\_work\pytorch\pytorch\test>if "" == "1" (
curl --retry 3 -k pytorch/test-infra@main/stats/windows_smoke_tests.csv (raw) --output .pytorch_specified_test_cases.csv  
 if ERRORLEVEL 1 exit /b 1  
 python run_test.py --exclude-jit-executor --shard 2 2 --verbose --run-specified-test-cases 
)  else (python run_test.py --exclude-jit-executor --shard 2 2 --verbose ) 
No CUDA runtime is found, using CUDA_HOME='C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.5'
```

From https://github.com/pytorch/pytorch/runs/4515996378?check_suite_focus=true

**Pull request closed without merging**